common-react/@dbeaver/ui-kit/src/Button/ButtonRoot.tsx

```diff
@@ -0,0 +1,21 @@
+import { Button as AriaButton, type ButtonProps as AriaKitButtonProps } from '@ariakit/react';
+import { Spinner } from '../Spinner/Spinner.js';
+
+export interface ButtonPropsWithLoader extends Omit<AriaKitButtonProps, 'clickOnEnter' | 'clickOnSpace'> {
+  loading?: boolean;
+  loader?: React.ReactNode;
+}
+
+export function ButtonRoot({ loading, loader, children, onClick, ...props }: ButtonPropsWithLoader) {
```

Comment:
It's common to have matching names for a component and its props. I recommend renaming `ButtonPropsWithLoader` to `ButtonRootProps`